ui_framework/src/components/context_menu/context_menu.js

```diff
@@ -27,55 +29,95 @@ export class KuiContextMenu extends Component {
     super(props);
 
     this.resetTransitionTimeout = undefined;
+    this.menuItems = [];
 
     this.state = {
       outGoingPanelId: undefined,
       currentPanelId: props.initialPanelId,
       transitionDirection: undefined,
+      isTransitioning: false,
+      focusedMenuItemIndex: 0,
     };
   }
 
+  onKeyDown = e => {
+    switch (e.keyCode) {
+      case cascadingMenuKeyCodes.UP:
+        if (this.menuItems.length) {
+          e.preventDefault();
+          const nextFocusedMenuItemIndex = this.state.focusedMenuItemIndex - 1;
+
+          this.setState({
+            focusedMenuItemIndex: nextFocusedMenuItemIndex < 0 ? this.menuItems.length - 1 : nextFocusedMenuItemIndex,
+          });
+        }
+        break;
+
+      case cascadingMenuKeyCodes.DOWN:
+        if (this.menuItems.length) {
+          e.preventDefault();
+          const nextFocusedMenuItemIndex = this.state.focusedMenuItemIndex + 1;
+
+          this.setState({
+            focusedMenuItemIndex: nextFocusedMenuItemIndex > this.menuItems.length - 1 ? 0 : nextFocusedMenuItemIndex,
+          });
+        }
+        break;
+
+      case cascadingMenuKeyCodes.LEFT:
+        e.preventDefault();
+        this.showPreviousPanel();
+        break;
+
+      case cascadingMenuKeyCodes.RIGHT:
+        if (this.menuItems.length) {
+          e.preventDefault();
+          this.menuItems[this.state.focusedMenuItemIndex].click();
+        }
+        break;
+
+      default:
+        break;
+    }
+  };
+
   showPanel(panelId, direction) {
     clearTimeout(this.resetTransitionTimeout);
 
     this.setState({
       outGoingPanelId: this.state.currentPanelId,
       currentPanelId: panelId,
       transitionDirection: direction,
+      isTransitioning: true,
     });
 
     // Queue the transition to reset.
     this.resetTransitionTimeout = setTimeout(() => {
       this.setState({
         transitionDirection: undefined,
+        isTransitioning: false,
+        focusedMenuItemIndex: 0,
       });
     }, 250);
   }
 
+  showPreviousPanel = () => {
+    const previousPanelId = this.props.idToPreviousPanelIdMap[this.state.currentPanelId];
```

Comment:
There is actually a bug, that you can press <kbd>Left</kbd> in the topmost menu and will receive an empty menu. I assume a check here would be the fix.